java/org/apache/coyote/http11/Http11InputBuffer.java

```diff
@@ -723,7 +723,7 @@ private boolean fill(boolean block) throws IOException {
                     // Avoid unknown protocol triggering an additional error
                     request.protocol().setString(Constants.HTTP_11);
                 }
-                throw new IllegalArgumentException(sm.getString("iib.requestheadertoolarge.error"));
+                throw new HeadersTooLargeException(sm.getString("iib.requestheadertoolarge.error"));
             }
```

Comment:
This will trigger both if the request line is too long (OK) and if the HTTP headers are too big (bad).

Reply:
What about https://tools.ietf.org/html/rfc6585#section-5?

Reply:
> This will trigger both if the request line is too long (OK) and if the HTTP headers are too big (bad).



> What about https://tools.ietf.org/html/rfc6585#section-5?

How about if I change it to 431

Reply:
This looks incomplete to me. What about https://github.com/apache/tomcat/blob/f4e7cdcff5abc239866e1d69906086cf1bae78ad/java/org/apache/coyote/http11/Http11Processor.java#L379-L389?

Reply:
If a more specific error code that 400 is used here (400 is an acceptable error code for both error conditions) then the right more specific error code needs to be used for each error condition.